src/components/modals/ModalUserEdit.tsx

```diff
@@ -0,0 +1,363 @@
+            icon={<Mail />}
+            placeholder={t("settings.emailplaceholder")}
+            name="email"
+            value={userEmail}
+            error={userEmailError}
+            onChange={event => {
+              userToEdit.email = event.currentTarget.value;
+              setUserEmail(userToEdit.email);
+              validateEmail(userToEdit.email);
+            }}
+          />
+          <TextInput
+            label={t("settings.firstname")}
+            icon={<User />}
+            placeholder={t("settings.firstnameplaceholder")}
+            name="firstname"
+            value={userFirst}
+            onChange={event => {
+              userToEdit.first_name = event.currentTarget.value;
+              setUserFirst(userToEdit.first_name);
+            }}
+          />
+          <TextInput
+            label={t("settings.lastname")}
+            icon={<User />}
+            placeholder={t("settings.lastnameplaceholder")}
+            name="lastname"
+            value={userLast}
+            onChange={event => {
+              userToEdit.last_name = event.currentTarget.value;
+              setUserLast(userToEdit.last_name);
+            }}
+          />
+        </SimpleGrid>
+        <PasswordEntry createNew={createNew} onValidate={onPasswordValidate} closing={closing} />
+      </Box>
+
+      <Title order={5}>{t("modalscandirectoryedit.header")} </Title>
+      <Text size="sm" color="dimmed">
+        {t("modalscandirectoryedit.explanation1")} &quot;{userToEdit ? userToEdit.username : "..."}&quot;{" "}
+        {t("modalscandirectoryedit.explanation2")}
+      </Text>
+      <Space h="md" />
+      <Grid grow>
+        <Grid.Col span={9}>
+          <TextInput
+            label={t("modalscandirectoryedit.currentdirectory")}
+            labelProps={{ style: { fontWeight: "bold" } }}
+            ref={inputRef}
+            error={pathError}
+            required={firstTimeSetup}
+            onChange={v => {
+              setNewScanDirectory(v.currentTarget.value);
+              validatePath(v.currentTarget.value);
+            }}
+            placeholder={scanDirectoryPlaceholder}
+          />
+        </Grid.Col>
+      </Grid>
+      <Title order={6}>{t("modalscandirectoryedit.explanation3")}</Title>
+      <div style={{ height: "150px", overflow: "auto" }}>
+        <SortableTree
+          innerStyle={{ outline: "none" }}
+          canDrag={() => false}
+          canDrop={() => false}
+          treeData={treeData}
+          onChange={changedTreeData => setTreeData(changedTreeData)}
+          theme={FileExplorerTheme}
+          isVirtualized={false}
+          generateNodeProps={rowInfo => {
+            const nodeProps = {
+              onClick: event => nodeClicked(event, rowInfo),
+            };
+            if (selectedNodeId === rowInfo.node.id) {
+              // @ts-ignore
+              nodeProps.className = "selected-node";
+            }
+            return nodeProps;
+          }}
```

Comment:
Again, these functions are redefined on every render. I'm not sure that it's the best approach.

Reply:
This is code from the legacy component. So this would be an optional fix from my point of view.